sdk/cognitivelanguage/azure-ai-language-questionanswering/samples/authoring/sample_update_knowledge_sources.py

```diff
@@ -0,0 +1,133 @@
+"""
+
+def sample_update_knowledge_sources():
+    # [START update_knowledge_sources]
+    import os
+    from azure.core.credentials import AzureKeyCredential
+    from azure.ai.language.questionanswering.projects import QuestionAnsweringProjectsClient
+
+    # get service secrets
+    endpoint = os.environ["AZURE_QUESTIONANSWERING_ENDPOINT"]
+    key = os.environ["AZURE_QUESTIONANSWERING_KEY"]
+
+    # create client
+    client = QuestionAnsweringProjectsClient(endpoint, AzureKeyCredential(key))
+    with client:
+
+        # create project
+        project_name = "Microsoft"
+        client.create_project(
+            project_name=project_name,
+            options={
+                "description": "test project for some Microsoft QnAs",
+                "language": "en",
+                "multilingualResource": True,
+                "settings": {
+                    "defaultAnswer": "no answer"
+                }
+            })
+
+        # sources
+        sources_poller = client.begin_update_sources(
+            project_name=project_name,
+            sources=[{
+                "op": "add",
+                "value": {
+                    "displayName": "MicrosoftFAQ",
+                    "source": "https://www.microsoft.com/en-in/software-download/faq",
+                    "sourceUri": "https://www.microsoft.com/en-in/software-download/faq",
+                    "sourceKind": "url",
+                    "contentStructureKind": "unstructured",
+                    "refresh": False
+                }
+            }]
+        )
+        sources_poller.result() # wait until done
+
+        sources = client.list_sources(
+            project_name=project_name
+        )
+        for item in sources:
+            print(u"source name: {}".format(item["displayName"]))
+            print(u"\tsource: {}".format(item["source"]))
+            print(u"\tsource uri: {}".format(item["sourceUri"]))
+            print(u"\tsource kind: {}".format(item["sourceKind"]))
+
+        # qnas
+        qna_poller = client.begin_update_qnas(
+            project_name=project_name,
+            qnas=[{
+                "op": "add",
+                "value": {
+                    "questions": [
+                        "What is the easiest way to use azure services in my .NET project?"
+                    ],
+                    "answer": "Using Microsoft's Azure SDKs"
+                }
+            }]
+        )
+        qna_poller.result()
+
+        qnas = client.list_qnas(
+            project_name=project_name
+        )
+        for item in qnas:
+            print(u"qna: {}".format(item["id"]))
+            print("\tquestions:")
+            for question in item["questions"]:
+                print(u"\t\t{}".format(question))
+            print(u"\tanswer: {}".format(item["answer"]))
+
+        # synonyms
+        client.update_synonyms(
+            project_name=project_name,
+            synonyms={
+                "value": [
```

Comment:
Yeah it's kinda weird that this payload is so different to the other update operations

Reply:
addressed this [here](https://github.com/Azure/azure-sdk-for-python/pull/21803#discussion_r795211708)